src/main/java/de/tum/cit/aet/artemis/iris/service/settings/IrisSettingsService.java

```diff
@@ -276,9 +300,135 @@ private IrisCourseSettings updateCourseSettings(IrisCourseSettings existingSetti
         ));
         // @formatter:on
 
+        // Automatically update the exercise settings when the enabledForCategories is changed
+        var newEnabledForCategoriesExerciseChat = existingSettings.getIrisChatSettings() == null ? new TreeSet<String>()
+                : existingSettings.getIrisChatSettings().getEnabledForCategories();
+        if (!Objects.equals(oldEnabledForCategoriesExerciseChat, newEnabledForCategoriesExerciseChat)) {
+            programmingExerciseRepository.findAllWithCategoriesByCourseId(existingSettings.getCourse().getId())
+                    .forEach(exercise -> setEnabledForExerciseByCategories(exercise, oldEnabledForCategoriesExerciseChat, newEnabledForCategoriesExerciseChat));
```

Comment:
Can you reduce this query to the categories that either got added or removed?

Reply:
Due to the way categories are saved in the database, this is currently not feasible. See #9469 for some details.